test/TwinGet.AutomationInterface.Test/TestUtils/TestTwincatProject.cs

```diff
@@ -0,0 +1,39 @@
+﻿// This file is licensed to you under MIT license.
+
+using System.Xml.Serialization;
+using TwinGet.AutomationInterface.Exceptions;
+using TwinGet.AutomationInterface.ProjectFileDeserialization;
+
+namespace TwinGet.AutomationInterface.Test.TestUtils
+{
+    internal class TestTwincatProject
+    {
+        private readonly List<TestPlcProject> _plcProjects = [];
+        public string Name { get; }
+        public string AbsolutePath { get; }
+        public IReadOnlyList<TestPlcProject> PlcProjects { get => _plcProjects; }
+
+        public TestTwincatProject(string name, string absolutePath)
+        {
+            Name = name;
+            AbsolutePath = absolutePath;
+
+            string xmlContent = File.ReadAllText(absolutePath);
+            XmlSerializer serializer = new(typeof(TcSmProjectData));
+            string? rootDir = Path.GetDirectoryName(absolutePath);
+
+            using (StringReader reader = new(xmlContent))
+            {
+                TcSmProjectData tcSmProject = serializer.Deserialize(reader) as TcSmProjectData ?? throw new InvalidProjectFileFormat("The format of TwinCAT project file is invalid.", absolutePath);
+
+                if (tcSmProject.Project.Plc?.Projects is not null)
+                {
+                    _plcProjects.AddRange(from ProjectElement project in tcSmProject.Project.Plc.Projects
```

Comment:
Could this be done in LINQ?
tcSmProject.Project.Plc.Projects.Select(x => x.plcProject is not null).FirstOrDefault() or something like that?

Reply:
Let's try again

```C#
var plcProjects = tcSmProject.Project.Plc?.Projects?
                .Select(x => new TestPlcProject(Path.Join(rootDir, x.PrjFilePath)));

_plcProjects = new(plcProjects ?? []);
```

How about this?
Lines 29-35 are removed and replaced with the above.
Then also get rid of the initialization for `_plcProjects` in line 11, and instead, initialization is in the constructor with the code above.

Reply:
Much better, ready for merge on my behalf

Reply:
Code changed to the mentioned snippet below,

> ```cs
> var plcProjects = tcSmProject.Project.Plc?.Projects?
>                 .Select(x => new TestPlcProject(Path.Join(rootDir, x.PrjFilePath)));
> 
> _plcProjects = new(plcProjects ?? []);
> ```
> 
> How about this? Lines 29-35 are removed and replaced with the above. Then also get rid of the initialization for `_plcProjects` in line 11, and instead, initialization is in the constructor with the code above.